protocols/protocol.py

```diff
@@ -117,15 +146,122 @@ def extended_validation(self):
                     return False
         return self.validate(self.toJsonDict())
 
-
     @classmethod
-    def validate(cls, jsonDict):
+    def validate(cls, jsonDict, debug=False):
         """
         Validates the specified JSON dictionary to determine if it is an
         instance of this element's schema.
         """
-        return avro.io.validate(cls.schema, jsonDict)
+        if debug:
+            validation_result = ValidationResult()
+            return cls.validate_debug(jsonDict=jsonDict, validation_result=validation_result)
+        return avro.io.validate(expected_schema=cls.schema, datum=jsonDict)
 
+    @classmethod
+    def validate_debug(cls, jsonDict, validation_result, expected_schema=None):
+        """
+        Returns ValidationResult with fields:
+                - result (True or False)
+                - messages (List of message strings ideally to help debug the problem)
+        """
+        INT_MIN_VALUE = -(1 << 31)
+        INT_MAX_VALUE = (1 << 31) - 1
+        LONG_MIN_VALUE = -(1 << 63)
```

Comment:
max and min long values in Python seem to be limited by memory as they use arbitrary-precision arithmetic (https://en.wikipedia.org/wiki/Arbitrary-precision_arithmetic) as opposed to fixes-precision arithmetic used for ints.
The value you set should be OK though.